### PR TITLE
[testing] get JS testing working

### PR DESCRIPTION
**Pull request by ascott**
- gets `npm run test` working
- adds test for `QueryAndSaveButtons`
- moves dev dependencies to `devDependencies` in package.json
- updates description in package.json

---
#### todo
- [x] get npm run test working with travis ci

plz review @mistercrunch @georgeke @williaster

### CODE REVIEW

File: caravel/assets/spec/javascripts/explore/components/QueryAndSaveBtns_spec.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  it('renders', () => {
+    expect(React.isValidElement(<QueryAndSaveButtons {...defaultProps} />)).to.equal(true);
+  });
+
+  // Test the output
+  describe('output', () => {
+    let wrapper;
+
+    beforeEach(() => {
+      wrapper = shallow(<QueryAndSaveButtons {...defaultProps} />);
```
Comment (ascott) [inline caravel/assets/spec/javascripts/explore/components/QueryAndSaveBtns_spec.jsx]: hi @lelandrichardson! :wave: i'm just getting enzyme setup in this repo and i'm having trouble with `shallow` rendering. when i run this spec, i get the following error. any ideas? 

screenshot 2016-07-13 21 38 30

**Comment by mistercrunch** [comment]
Nice! The entry point for Travis to run the js build (as its own item in the build matrix) is this bash script:
https://github.com/airbnb/caravel/blob/master/caravel/assets/js_build.sh

I think you want to add `npm run test` to the list and `npm run prod` may or may not be necessary anymore depending on what `npm run test` does...

File: caravel/assets/spec/javascripts/explore/components/QueryAndSaveBtns_spec.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+var React = require('react');
```
Comment (williaster) [inline caravel/assets/spec/javascripts/explore/components/QueryAndSaveBtns_spec.jsx]: just curious what's the advantage of using `var` for packages vs `const` ? or `import`ing them?

**Comment by williaster** [comment]
nice!!!! agree adding it to travis would be a good move, and figuring out the enzyme shallow render issue. 

not sure how max/george feel but I'd <3 a short overview of the testing component setup (mainly the things in `spec/setup.js`, I think the `_spec` files themselves are pretty self-explanatory). I took leland's testing class and the setup part wasn't covered.

File: caravel/assets/spec/javascripts/explore/components/QueryAndSaveBtns_spec.jsx (hunk 1 above)
Comment (ascott) [inline caravel/assets/spec/javascripts/explore/components/QueryAndSaveBtns_spec.jsx]: resolved this, wasn't importing the component into the spec properly.

**Comment by ascott** [comment]
@willaster i can give a short overview during team meeting on monday or just whenever. i renamed `spec/setup.js` to `spec/browser.js`, it's mostly borrowed from the enzyme docs, but essentially it's needed to use jsdom as a headless browser for the testing env. there's more info here: https://github.com/airbnb/enzyme/blob/master/docs/guides/jsdom.md#using-enzyme-with-jsdom.

**Pull request merged**